tests/util/CMakeLists.txt

```diff
@@ -44,7 +44,7 @@ set_target_properties(util_tests PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${libgit2_B
 
 target_include_directories(util_tests PRIVATE ${TEST_INCLUDES} ${LIBGIT2_INCLUDES} ${LIBGIT2_DEPENDENCY_INCLUDES})
 target_include_directories(util_tests SYSTEM PRIVATE ${LIBGIT2_SYSTEM_INCLUDES})
-target_link_libraries(util_tests ${LIBGIT2_SYSTEM_LIBS})
+target_link_libraries(util_tests ${LIBGIT2_SYSTEM_LIBS} libgit2)
```

Comment:
The util tests should only depend on the util library; they shouldn't need libgit2 functions.

Reply:
tests/util/init.c needs git_libgit2_init git_libgit2_shutdown for example which seem to be inside libgit2.c

/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_git_sandbox_init':
/home/user/libgit2/tests/clar/clar_libgit2.c:207:(.text+0x9b3): undefined reference to `git_repository_open'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:210:(.text+0xa02): undefined reference to `git_repository_reinit_filesystem'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_git_sandbox_init_new':
/home/user/libgit2/tests/clar/clar_libgit2.c:217:(.text+0xa6d): undefined reference to `git_repository_init'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_git_sandbox_reopen':
/home/user/libgit2/tests/clar/clar_libgit2.c:226:(.text+0xada): undefined reference to `git_repository_free'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:229:(.text+0xb0b): undefined reference to `git_repository_open'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_git_sandbox_cleanup':
/home/user/libgit2/tests/clar/clar_libgit2.c:239:(.text+0xb69): undefined reference to `git_repository_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_commit_from_index':
/home/user/libgit2/tests/clar/clar_libgit2.c:422:(.text+0x154b): undefined reference to `git_revparse_ext'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:425:(.text+0x1569): undefined reference to `git_repository_index'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:426:(.text+0x15c6): undefined reference to `git_index_write_tree'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:427:(.text+0x1619): undefined reference to `git_index_write'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:428:(.text+0x1667): undefined reference to `git_index_free'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:430:(.text+0x168c): undefined reference to `git_tree_lookup'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:435:(.text+0x176b): undefined reference to `git_signature_now'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:437:(.text+0x17e2): undefined reference to `git_signature_new'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:442:(.text+0x187a): undefined reference to `git_oid_tostr'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:447:(.text+0x18d5): undefined reference to `git_reference_name'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:447:(.text+0x1911): undefined reference to `git_commit_create_v'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:452:(.text+0x1977): undefined reference to `git_oid_cpy'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:454:(.text+0x1986): undefined reference to `git_object_free'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:455:(.text+0x1995): undefined reference to `git_reference_free'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:457:(.text+0x19ad): undefined reference to `git_signature_free'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:458:(.text+0x19bc): undefined reference to `git_tree_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_set_bool':
/home/user/libgit2/tests/clar/clar_libgit2.c:464:(.text+0x1a1a): undefined reference to `git_repository_config'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:465:(.text+0x1a72): undefined reference to `git_config_set_bool'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:466:(.text+0x1ab4): undefined reference to `git_config_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_get_bool':
/home/user/libgit2/tests/clar/clar_libgit2.c:473:(.text+0x1b09): undefined reference to `git_repository_config'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:474:(.text+0x1b56): undefined reference to `git_config_get_bool'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:476:(.text+0x1b6b): undefined reference to `git_config_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_set_int':
/home/user/libgit2/tests/clar/clar_libgit2.c:483:(.text+0x1bbe): undefined reference to `git_repository_config'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:484:(.text+0x1c0f): undefined reference to `git_config_set_int32'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:485:(.text+0x1c51): undefined reference to `git_config_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_get_int':
/home/user/libgit2/tests/clar/clar_libgit2.c:492:(.text+0x1ca6): undefined reference to `git_repository_config'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:493:(.text+0x1cf3): undefined reference to `git_config_get_int32'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:495:(.text+0x1d08): undefined reference to `git_config_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_repo_set_string':
/home/user/libgit2/tests/clar/clar_libgit2.c:502:(.text+0x1d5c): undefined reference to `git_repository_config'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:503:(.text+0x1dae): undefined reference to `git_config_set_string'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:504:(.text+0x1df0): undefined reference to `git_config_free'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_fake_homedir_cleanup':
/home/user/libgit2/tests/clar/clar_libgit2.c:582:(.text+0x224e): undefined reference to `git_libgit2_opts'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:583:(.text+0x2293): undefined reference to `git_buf_dispose'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_fake_homedir':
/home/user/libgit2/tests/clar/clar_libgit2.c:591:(.text+0x22ea): undefined reference to `git_libgit2_opts'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:600:(.text+0x2438): undefined reference to `git_libgit2_opts'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_fake_globalconfig_cleanup':
/home/user/libgit2/tests/clar/clar_libgit2.c:619:(.text+0x2541): undefined reference to `git_libgit2_opts'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:620:(.text+0x2586): undefined reference to `git_buf_dispose'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_fake_globalconfig':
/home/user/libgit2/tests/clar/clar_libgit2.c:628:(.text+0x25e2): undefined reference to `git_libgit2_opts'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:637:(.text+0x2735): undefined reference to `git_libgit2_opts'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_sandbox_set_homedir':
/home/user/libgit2/tests/clar/clar_libgit2.c:650:(.text+0x27fb): undefined reference to `git_libgit2_opts'
/usr/bin/ld: /home/user/libgit2/tests/clar/clar_libgit2.c:657:(.text+0x2891): undefined reference to `git_libgit2_opts'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o: in function `cl_sandbox_set_search_path_defaults':
/home/user/libgit2/tests/clar/clar_libgit2.c:672:(.text+0x297c): undefined reference to `git_libgit2_opts'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2.c.o:/home/user/libgit2/tests/clar/clar_libgit2.c:674: more undefined references to `git_libgit2_opts' follow
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2_trace.c.o: in function `git_trace__write_fmt':
/home/user/libgit2/src/libgit2/trace.h:27:(.text+0x36): undefined reference to `git_trace__data'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2_trace.c.o: in function `git_trace':
/home/user/libgit2/src/libgit2/trace.h:41:(.text+0x121): undefined reference to `git_trace__data'
/usr/bin/ld: /home/user/libgit2/src/libgit2/trace.h:42:(.text+0x132): undefined reference to `git_trace__data'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2_trace.c.o: in function `cl_global_trace_register':
/home/user/libgit2/tests/clar/clar_libgit2_trace.c:241:(.text+0x681): undefined reference to `git_trace_set'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/clar_libgit2_trace.c.o: in function `cl_global_trace_disable':
/home/user/libgit2/tests/clar/clar_libgit2_trace.c:255:(.text+0x6c2): undefined reference to `git_trace_set'
/usr/bin/ld: CMakeFiles/util_tests.dir/__/clar/main.c.o: in function `main':
/home/user/libgit2/tests/clar/main.c:19:(.text+0x21): undefined reference to `git_libgit2_init'
/usr/bin/ld: /home/user/libgit2/tests/clar/main.c:38:(.text+0xb8): undefined reference to `git_libgit2_shutdown'
/usr/bin/ld: CMakeFiles/util_tests.dir/init.c.o: in function `test_init__returns_count':
/home/user/libgit2/tests/util/init.c:8:(.text+0x5): undefined reference to `git_libgit2_init'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:9:(.text+0x4b): undefined reference to `git_libgit2_init'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:11:(.text+0x91): undefined reference to `git_libgit2_shutdown'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:12:(.text+0xd7): undefined reference to `git_libgit2_shutdown'
/usr/bin/ld: CMakeFiles/util_tests.dir/init.c.o: in function `test_init__reinit_succeeds':
/home/user/libgit2/tests/util/init.c:17:(.text+0x124): undefined reference to `git_libgit2_shutdown'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:18:(.text+0x16a): undefined reference to `git_libgit2_init'
/usr/bin/ld: CMakeFiles/util_tests.dir/init.c.o: in function `reinit':
/home/user/libgit2/tests/util/init.c:28:(.text+0x1d0): undefined reference to `git_libgit2_init'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:29:(.text+0x20e): undefined reference to `git_libgit2_shutdown'
/usr/bin/ld: CMakeFiles/util_tests.dir/init.c.o: in function `test_init__concurrent_init_succeeds':
/home/user/libgit2/tests/util/init.c:42:(.text+0x277): undefined reference to `git_libgit2_init'
/usr/bin/ld: /home/user/libgit2/tests/util/init.c:49:(.text+0x31f): undefined reference to `git_libgit2_shutdown'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/util/CMakeFiles/util_tests.dir/build.make:1127: util_tests] Error 1
make[1]: *** [CMakeFiles/Makefile2:523: tests/util/CMakeFiles/util_tests.dir/all] Error 2
make: *** [Makefile:146: all] Error 2